### PR TITLE
ux of data-fetching exercise

### PR DESCRIPTION
**Pull request by arthurpankiewicz**

### CODE REVIEW

File: ux-of-data-fetching/Exercise1/problem/App.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          <tr>
+            <th>Name</th>
+            <th>Age</th>
+            <th>Occupation</th>
+          </tr>
+        </thead>
+        <tbody>
+          {!data || data.length === 0
+            ? null
+            : data.map((person: any) => (
```
Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/problem/App.tsx:79]: `any` -> `Person`

Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/problem/App.tsx]: `any` -> `Person`

File: ux-of-data-fetching/Exercise1/solution/App.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          <tr>
+            <th>Name</th>
+            <th>Age</th>
+            <th>Occupation</th>
+          </tr>
+        </thead>
+        <tbody>
+          {error || !data || data.length === 0
+            ? null
+            : data.map((person: any) => (
```
Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/solution/App.tsx]: `any` -> `Person`

File: ux-of-data-fetching/Exercise1/problem/App.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      .then((data) => setData(data));
+  }
+
+  useEffect(() => {
+    fetchData("/api/people");
+  }, []);
+
+  return (
+    <>
+      <div>
```
Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/problem/App.tsx]: Maybe `nav` instead?

File: ux-of-data-fetching/Exercise1/problem/App.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>Age</th>
+            <th>Occupation</th>
+          </tr>
+        </thead>
+        <tbody>
+          {!data || data.length === 0
+            ? null
```
Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/problem/App.tsx:78]: Out of curiosity: is there a reason not to do something like `data && data.length > 0 && data.map`…?

File: ux-of-data-fetching/Exercise1/problem/App.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          Simulate empty
+        </button>
+      </div>
+
+      <table>
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>Age</th>
+            <th>Occupation</th>
```
Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/problem/App.tsx]: I’m not 100% sure it’s necessary, but in general I always write `<th scope="col">` to make it explicit that the cells are column headers.

File: ux-of-data-fetching/Exercise1/solution/styles.css [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  border-radius: 4px;
+  cursor: pointer;
+  transition: background-color 0.2s ease;
+  margin: 10px;
+}
+button:hover {
+  background-color: #1199e6;
+}
+button:focus {
+  outline: none;
```
Comment (chasenlehara) [inline ux-of-data-fetching/Exercise1/solution/styles.css:74]: No focus indicator?

**Review by chasenlehara — APPROVED** [review]
The solution didn’t load for me in CodeSandbox, so I just did a light review of the code. I’d like to take a look again when it’s working. Let me know if I can help figure out why CodeSandbox isn’t working.

File: ux-of-data-fetching/Exercise1/solution/styles.css (hunk 6 above)
Comment (arthurpankiewicz) [reply ux-of-data-fetching/Exercise1/solution/styles.css:74]: lol not sure why i have this in here, got rid of it

File: ux-of-data-fetching/Exercise1/problem/App.tsx (hunk 4 above)
Comment (arthurpankiewicz) [reply ux-of-data-fetching/Exercise1/problem/App.tsx:78]: your way is better, but we won't get a crash on error which I want to show for the starting point 😄

**Pull request merged**